### PR TITLE
Update dependencies for the template

### PR DESCRIPTION
**Pull request by rahulPalliyalil**
### What

Update template dependencies . The changes include:

- Update to version 3 of kafka library
- Update to version 2 of the dp net library

### How to review

The repos for api, base-app, controller, event, generic and library has been generated and tested. 

### Who can review

Anyone but me

### CODE REVIEW

File: project_generation/content/templates/event/cmd/producer/main.go.tmpl [hunk 1]
```diff
@@ -40,14 +38,14 @@ func main() {
...
-	kafkaProducer, err := kafka.NewProducer(ctx, cfg.KafkaConfig.Brokers, cfg.KafkaConfig.HelloCalledTopic, pChannels, pConfig)
+	kafkaProducer, err := kafka.NewProducer(ctx, pConfig)
 	if err != nil {
 		log.Fatal(ctx, "fatal error trying to create kafka producer", err, log.Data{"topic": cfg.KafkaConfig.HelloCalledTopic})
 		os.Exit(1)
 	}
 
 	// kafka error logging go-routines
-	kafkaProducer.Channels().LogErrors(ctx, "kafka producer")
+    kafkaProducer.LogErrors(ctx)
```
Comment (nshumoogum) [inline project_generation/content/templates/event/cmd/producer/main.go.tmpl:48]: Missing extra indentation

File: project_generation/content/templates/event/service/service.go.tmpl [hunk 2]
```diff
@@ -47,8 +47,8 @@ func Run(ctx context.Context, serviceList *ExternalServiceList, buildTime, gitCo
 	// Event Handler for Kafka Consumer
 	event.Consume(ctx, consumer, &event.HelloCalledHandler{}, cfg)
 
-	// Kafka error logging go-routine
-	consumer.Channels().LogErrors(ctx, "kafka consumer")
+		// Kafka error logging go-routine
```
Comment (nshumoogum) [inline project_generation/content/templates/event/service/service.go.tmpl:50]: Too much indentation this time

File: project_generation/content/templates/event/service/service.go.tmpl [hunk 3]
```diff
@@ -104,7 +104,7 @@ func (svc *Service) Close(ctx context.Context) error {
 		// The kafka consumer will be closed after the service shuts down.
 		if svc.serviceList.KafkaConsumer {
 			log.Info(ctx, "stopping kafka consumer listener")
-			if err := svc.consumer.StopListeningToConsumer(ctx); err != nil {
+			if err := svc.consumer.StopAndWait(); err != nil {
```
Comment (nshumoogum) [inline project_generation/content/templates/event/service/service.go.tmpl:107]: I believe we should just use the Stop method instead of StopAndWait

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: project_generation/content/templates/event/service/service.go.tmpl (hunk 3 above)
Comment (rahulPalliyalil) [reply project_generation/content/templates/event/service/service.go.tmpl:107]: From the readme , it seems the stopandwait as the replacement for StopListeningToConsumer. please let me know if this is not the case.

File: project_generation/content/templates/event/cmd/producer/main.go.tmpl (hunk 1 above)
Comment (rahulPalliyalil) [reply project_generation/content/templates/event/cmd/producer/main.go.tmpl]: Fixed in update 1.

File: project_generation/content/templates/event/service/service.go.tmpl (hunk 2 above)
Comment (rahulPalliyalil) [reply project_generation/content/templates/event/service/service.go.tmpl]: Fixed in update 1.

File: project_generation/content/templates/event/service/service.go.tmpl (hunk 3 above)
Comment (rahulPalliyalil) [reply project_generation/content/templates/event/service/service.go.tmpl]: done.

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**